### PR TITLE
syslog module lokahi opennms

### PR DESCRIPTION
**Pull request by MusaidAli**
## Description
SysLog implementation module

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-XXXX

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

**Review by cgorantla — COMMENTED** [review]
Not sure what you are trying to do here.

Minion is an OSGi application, it runs on karaf, there is no need to for any spring boot dependencies.

Ignore the test cases,  Integrate SyslogReceiver , it's implementation along with blueprint and feature files. Make sure that this feature resolves properly on karaf when we run Minion.

File: minion/minion-syslog/features/src/main/feature/feature.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<features name="minion-syslog-${project.version}" xmlns="http://karaf.apache.org/xmlns/features/v1.6.0">
+
+    <feature name="minion-syslog" version="${project.version}">
+
+      <!--  <bundle>mvn:org.opennms.lokahi.minion/syslog-listener/${project.version}</bundle>-->
```
Comment (cgorantla) [inline minion/minion-syslog/features/src/main/feature/feature.xml:6]: you need to have required bundles for this feature.

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyncDispatcher.java
Comment (cgorantla) [inline minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyncDispatcher.java:1]: not required

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SinkModule.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
+/*
```
Comment (cgorantla) [inline minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SinkModule.java:1]: not required

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SimpleNamespaceFilter.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
+/*
```
Comment (cgorantla) [inline minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SimpleNamespaceFilter.java:1]: not required

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/MessageDispatcherFactory.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
+/*
```
Comment (cgorantla) [inline minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/MessageDispatcherFactory.java:1]: not required

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/Message.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to The OpenNMS Group, Inc (TOG) under one or more
```
Comment (cgorantla) [inline minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/Message.java:2]: not required

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/AsyncPolicy.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
+/*
```
Comment (cgorantla) [inline minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/AsyncPolicy.java:1]: not required

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/AsyncDispatcher.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
+/*
```
Comment (cgorantla) [inline minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/AsyncDispatcher.java:1]: not required

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/AggregationPolicy.java [hunk 8]
```diff
@@ -0,0 +1,99 @@
+/*
```
Comment (cgorantla) [inline minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/AggregationPolicy.java:1]: not required

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/AbstractXmlSinkModule.java [hunk 9]
```diff
@@ -0,0 +1,103 @@
+/*
```
Comment (cgorantla) [inline minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/AbstractXmlSinkModule.java:1]: not required

File: minion/minion-syslog/listener/pom.xml [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+    <packaging>bundle</packaging>
+    <name>OpenNMS Lokahi :: Minion :: Syslog :: Listener</name>
+
+
+    <dependencies>
+
+
+
+        <!-- https://mvnrepository.com/artifact/org.apache.camel/camel-test -->
+        <dependency>
```
Comment (cgorantla) [inline minion/minion-syslog/listener/pom.xml:41]: clean up these libraries that are not required

**Review by cgorantla — CHANGES_REQUESTED** [review]
First get the Receiver loaded properly along with bundle and feature. Then we can add more layers

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/MessageDispatcherFactory.java (hunk 4 above)
Comment (MusaidAli) [reply minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/MessageDispatcherFactory.java:1]: m_dispatcher.send(new SyslogConnection(source, buffer)).get();
this is required.

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SimpleNamespaceFilter.java (hunk 3 above)
Comment (MusaidAli) [reply minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SimpleNamespaceFilter.java:1]: public static XMLFilter getXMLFilterForNamespace(final String namespace) throws SAXException {
        XMLFilter filter = namespace == null ? new SimpleNamespaceFilter("", false) : new SimpleNamespaceFilter(namespace, true);

        LOG.trace("namespace filter for namespace {}: {}", namespace, filter);
        final XMLReader xmlReader = XMLReaderFactory.createXMLReader();
        xmlReader.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
        xmlReader.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
        xmlReader.setFeature("http://xml.org/sax/features/external-general-entities", false);
        xmlReader.setFeature("http://xml.org/sax/features/external-parameter-entities", false);

        filter.setParent(xmlReader);
        return filter;
    }

this is required

**Pull request closed without merging**